<?xml version="1.0"?>
<robot name="visual">

<material name="blue">
  <color rgba="0 0 1 1"/>
</material>
<material name="gray">
  <color rgba="0.5 0.5 0.5 1"/>
</material>
<material name="black">
  <color rgba="0 0 0 1"/>
</material>
<material name="white">
  <color rgba="1 1 1 1"/>
</material>
<material name="red">
  <color rgba="1 0 0 1"/>
</material>
<material name="green">
  <color rgba="0 1 0 1"/>
</material>


<link name="base_link">
  <visual>
    <geometry>
      <cylinder length="0.1" radius="4"/>
    </geometry>
    <material name="white"/>
    <origin rpy="0 0 0" xyz="0 0 -0.05"/>
  </visual>
</link>

<link name="cylinder_1">
  <visual>
    <geometry>
      <cylinder length="1" radius="0.2"/>
    </geometry>
    <origin rpy="0 0 0" xyz="0 0 0.5"/>
    <material name="blue"/>
  </visual>
</link>

<joint name="base_link_cylinder_1" type="continuous">
  <parent link="base_link"/>
  <child link="cylinder_1"/>
  <axis xyz="0 0 1"/>
  <origin xyz="0 0 0.0"/>
</joint>

<link name="sphere_1">
    <visual>
      <geometry>
        <sphere radius="0.25"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
      <material name="gray"/>
    </visual>
</link>

<joint name="cylinder_1_sphere_1" type="fixed">
    <parent link="cylinder_1"/>
    <child link="sphere_1"/>
</joint>

<link name="cylinder_2">
    <visual>
      <geometry>
        <cylinder length="1" radius="0.2"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
      <material name="green"/>
    </visual>
 </link>

<joint name="cylinder_1_cylinder_2" type="revolute">
    <parent link="cylinder_1"/>
    <child link="cylinder_2"/>
    <axis xyz="0 1.0 0"/>
    <limit effort="1000.0" lower="-1.570" upper="1.570" velocity="1"/>
    <origin rpy="0 0 0" xyz="0 0 1.0"/>
  </joint>


<link name="sphere_2">
    <visual>
      <geometry>
        <sphere radius="0.25"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 1.0"/>
      <material name="gray"/>
    </visual>
</link>

<joint name="cylinder_2_sphere_2" type="fixed">
    <parent link="cylinder_2"/>
    <child link="sphere_2"/>
</joint>

<link name="cylinder_3">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.1"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.5"/>
      <material name="black"/>
    </visual>
 </link>

<joint name="cylinder_2_cylinder_3" type="revolute">
  <parent link="cylinder_2"/>
  <child link="cylinder_3"/>
  <limit effort="1000.0" lower="-1.570" upper="1.570" velocity="1"/>
  <axis xyz="0 1 0"/>
  <origin xyz="0 0 1.0"/>
</joint>

</robot>